### DISCUSSION TITLE
Render.com can't find public assets -> 404

### DISCUSSION THREAD

**Discussion by lockinmarv** [vikejs/vike#655] [Discussion]
Hey! I'm trying to deploy an SSR app with a Fastify server to Render.com but always encounter that some public resources can't be found. The Fastify server code is basically from the example repo (https://github.com/giacomorebonato/vite-plugin-ssr-fastify) + I added some additional code to handle auth, redirects and cookies.

Live site: https://makertools-app.onrender.com

Locally in development but also when previewing the generated files (`ts-node ./server/index.ts`) I don't get any errors.

Am I doing something wrong or is this a problem with Render.com?

Error in the console:
<img width="536" alt="CleanShot 2023-02-16 at 11 01 58@2x" src="https://user-images.githubusercontent.com/6840287/219333862-5097cf10-4590-4bb7-8f69-06992bdb5a9c.png">

Generated dist:
<img width="314" alt="CleanShot 2023-02-16 at 11 04 36@2x" src="https://user-images.githubusercontent.com/6840287/219334033-aa02ca33-fa9e-43d7-82c7-33b4f545a6c1.png">

**Comment by brillout** [comment]
I'm not familiar wiht render.com but have you configured render.com to statically serve the files living at `dist/client/`? That's something you need to do with other platforms such as Vercel or Cloudflare Workers.

**Reply by lockinmarv** [reply]
I think this could indeed the cause for this but Render doesn't let you define rewrites when using web services. You can only define it when using static sites. But I can't deploy the SSR app as static site because it needs a running server, right? 

I probably should use a different platform then. 🫠

**Reply by brillout** [reply]
I'd be surprise if Render doesn't support that use case 🤔. That's an almost ubiquitous use case.

**Reply by lockinmarv** [reply]
They support it but only for static sites (that's what I see and understand at least). I can define Rewrites & Redirects in the settings of static sites but not for web services. Pretty weird IMO.

https://render.com/docs/static-sites
https://render.com/docs/redirects-rewrites

![CleanShot 2023-02-16 at 13 41 48@2x](https://user-images.githubusercontent.com/6840287/219367994-ae7985d3-5792-466b-afbe-c322552ea920.png)
